src/main/java/dao/FileReaderImpl.java

```diff
@@ -0,0 +1,34 @@
+package dao;
+
+import java.io.BufferedReader;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import model.FruitTransaction;
+import model.OperationType;
+
+public class FileReaderImpl implements dao.FileReader {
```

Comment:
The FileReaderImpl is not following the encapsulation principle. Consider using a more restrictive access modifier for internal methods and data if they are not intended to be exposed or used outside the class.